### PR TITLE
Connect auth backend

### PR DESCRIPTION
**Pull request by darabahmed37**
It supersedes  the  [Basic Structure PR](https://github.com/darabahmed37/mytube-frontend/pull/1)

Connected with backend.

### CODE REVIEW

File: src/pages/Dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
+import React, { FC } from "react"
+import VideoGrid from "layouts/VideoGrid"
+
+const Dashboard: FC = () => {
+	return (
+		<>
+			<VideoGrid
```
Comment (Faizan-e-Muhammad) [inline src/pages/Dashboard.tsx:7]: Layouts should not be used like normal components.
Please update the layouts to only define layout of pages that have a common layout.

File: src/App.tsx [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
-          link: ""
-        }}/>
-      </BrowserRouter>
-    </ThemeProvider>
-  );
-};
+	return (
+		<ThemeProvider theme={theme}>
+			<BrowserRouter>
+				<Routes>
```
Comment (Faizan-e-Muhammad) [inline src/App.tsx:14]: Define App routes in a separate routes folder.

File: src/api/auth/index.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+export function getRefreshToken() {
+	return localStorage.getItem("refresh")
+}
+
+export async function signInWithEmailAndPassword(email: string, password: string): Promise<AxiosResponse> {
+	let response: AxiosResponse
+
+		response = await axios.post("auth/sign-in/", {
```
Comment (Faizan-e-Muhammad) [inline src/api/auth/index.ts]: Define all API routes in a separate routes file.

File: src/components/VideoCard.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+export interface VideoCardProps {
+	title: string
+	description: string
+	imageUrl: string
+}
+
+const VideoCard: FC<VideoCardProps> = ({ title, description, imageUrl }) => {
+	return (
+		<Card
+			sx={{
```
Comment (Faizan-e-Muhammad) [inline src/components/VideoCard.tsx]: Add custom styles in a separate styles file.

File: src/config/index.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+const { REACT_APP_BASE_BACKEND_URL } = process.env
+
+export const BASE_BACKEND_URL = REACT_APP_BASE_BACKEND_URL!.toString()
```
Comment (Faizan-e-Muhammad) [inline src/config/index.ts:3]: export const BASE_BACKEND_URL = ``${REACT_APP_BASE_BACKEND_URL}``

**Comment by darabahmed37** [comment]
- Closed In favour of [Refactoring ](https://github.com/darabahmed37/mytube-frontend/pull/3)

**Pull request closed without merging**